examples/borderrouter/windows/border-router.cpp

```
@@ -0,0 +1,379 @@
+// border-router.cpp : Defines the entry point for the console application.
+//
+
+#include "stdafx.h"
+#include <mbedtls/memory_buffer_alloc.h>
+#include <thread/thread_uris.hpp>
+#include <crypto/mbedtls.hpp>
+#include <memory>
+#include "border-router.hpp"
+#include "client.hpp"
+#include "fake-leader.hpp"
+
+#define MBED_MEMORY_BUF_SIZE  (2048 * sizeof(void*))
+
+extern "C" void otSignalTaskletPending(otInstance *)
+{
+}
+
+inline uint8_t _str1ToHex(const char charTuple) {
+    if ('0' <= charTuple && charTuple <= '9') {
+        return (uint8_t)(charTuple - '0');
+    }
+    else if ('A' <= charTuple && charTuple <= 'F') {
+        return (uint8_t)(10 + (charTuple - 'A'));
+    }
+    else if ('a' <= charTuple && charTuple <= 'f') {
+        return (uint8_t)(10 + (charTuple - 'a'));
+    }
+    return 0;
+}
+
+inline uint8_t _str2ToHex(const char hexByte[2]) {
+    const char hi = hexByte[0];
+    const char lo = hexByte[1];
+    return (_str1ToHex(hi) * 16) + _str1ToHex(lo);
+}
+
+void printBuffer(char* buffer, int len)
+{
+    for (int i = 0; i < len; i++)
+    {
+        printf("%02x", (unsigned char)*buffer++);
+        if (i % 4 == 3)
+        {
+            printf(" ");
+        }
+    }
+    printf("\n");
+}
+
+void getPSKc(const char* passPhrase, const char* networkName, const char* const xPanId, uint8_t* derivedKeyOut) {
```

This function should return an error instead of `void`.